daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -125,7 +136,7 @@ runLedgerUploadDar flags darPathM = do
         getDarPath
     putStrLn $ "Uploading " <> darPath <> " to " <> show hp
     bytes <- BS.readFile darPath
-    uploadDarFile hp bytes
+    uploadDarFile hp bytes `catch` \e -> throw (ExceptionDuringUploadDar e)
```

Comment:
Please use `catch` from `safe-exceptions` or even better `catchIO`. You don’t want to catch asynchronous exceptions here.

Reply:
I switch to safe-exceptions.
I couldn't use `catchIO` as this does not catch `GRPCIOTimeout` which is the one I really care about.